src/main/java/org/tron/program/Benchmark.java

```diff
@@ -0,0 +1,318 @@
+
+    TimeBenchmarkResult result = timeBenchmark();
+
+    return result;
+  }
+
+  public static long getMem() {
+
+    com.sun.management.OperatingSystemMXBean os = (com.sun.management.OperatingSystemMXBean)
+        java.lang.management.ManagementFactory.getOperatingSystemMXBean();
+    long physicalMemorySize = os.getTotalPhysicalMemorySize();
+
+    return (long)(physicalMemorySize * 1.0 / 1024 / 1024 * 0.8);
+  }
+
+  public static boolean checkJavaVersion() {
+    String name = System.getProperty("java.runtime.name").toLowerCase();
+    if (name.indexOf("openjdk") != -1) {
+      return false;
+    }
+    String[] version = System.getProperty("java.vm.specification.version").split("\\.");
+    if (version.length < 2) {
+      return false;
+    } else if (Integer.valueOf(version[0]) * 10000 + Integer.valueOf(version[1]) < 10008) {
+      return false;
+    } else {
+      return true;
+    }
+  }
+
+  public static String readFile(String fileName) {
+
+    File file = new File(fileName);
+    if (null != file && file.canRead()) {
+      try {
+        Reader in = new FileReader(file);
+        StringWriter out = new StringWriter();
+        copy(in, out);
+        return out.toString();
+      } catch(IOException e) {
+        logger.info(e.getMessage());
+      }
+    }
+    return null;
+  }
+
+  public static void writeFile(String content, String fileName) {
+
+    try {
+      File file = new File(fileName);
+      Writer writer = new FileWriter(file);
+      writer.write(content);
+      writer.close();
+    } catch (IOException e) {
+      logger.info(e.getMessage());
+    }
+  }
+
+  public static void main(String[] args) {
+
+    boolean checkVersion = checkJavaVersion();
+
+    if (checkVersion) {
+      System.out.println("1. JAVA VERSION:\nsatisfied");
+    } else {
+      System.out.println("1. JAVA VERSION:\njava MUST be oracle jdk, and version >= 1.8");
+    }
+
+    try {
+      long mem = getMem();
+      String content = readFile("start.sh");
+      String newContent = content.replaceAll("java -jar", "java -Xmx" + mem + "m -jar");
+      String newFileName = "build/libs/start-recommend.sh";
+      writeFile(newContent, newFileName);
+      System.out.println("2. MEMORY:\nwhen setup java, recommend to use: java -Xmx" + mem + "m\ncan see also " + newFileName);
+    } catch (Exception e) {
+      logger.info(e.getMessage());
+    }
+
+    initData();
+
+    TimeBenchmarkResult timeResult = null;
+    try {
+      timeResult = getTimeRatio();
+    } catch (ContractExeException | ReceiptCheckErrException | VMIllegalException | ContractValidateException e) {
+      logger.info(e.getMessage());
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      logger.info(e.getMessage());
+    }
+    if (null != timeResult) {
+      try {
+        String content = readFile("src/main/resources/config.conf");
+        String newContent = content
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: A "NullPointerException" could be thrown; "content" is nullable here. ([squid:S2259](https://sonarcloud.io/organizations/tron-zhaohong/rules#rule_key=squid%3AS2259))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=java-tron&pullRequest=1546&issues=AWXsmUACxJR_uy9B78-P&open=AWXsmUACxJR_uy9B78-P)

[ISSUE_KEY]: AWXsmUACxJR_uy9B78-P